File: docs/get-prices/request.py
'''
    Get-Prices RustOpus example
'''
from urllib import request

authcode: str = "AAAAAAAA-0000-0000-0000-BBBBBBBBBBBB"
pid: int = 1234
outfile: str = "get-prices.xml"

url: str = f"<url-to-rustopus>/get-prices?authcode={authcode}&pid={pid}"

with request.urlopen(url) as response:
    with open(file = outfile,
              mode = "wb") as f:
        f.write(response.read())

File: docs/get-prices/request.js
import { writeFile } from "fs/promises";

const AUTHCODE = "AAAAAAAA-0000-0000-0000-BBBBBBBBBBBB";
const PID = 1234;
const OUTFILE = "get-prices.xml";

const response = await fetch(`<url-to-rustopus>/get-prices?authcode=${AUTHCODE}&pid=${PID}`);
const text = await response.text();

await writeFile(OUTFILE, text);
